data-prepper-plugins/otlp-source/src/main/java/org/opensearch/dataprepper/plugins/source/otlp/OTLPSourceConfig.java

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.otlp;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.AssertTrue;
+import jakarta.validation.constraints.Size;
+import org.apache.commons.lang3.StringUtils;
+import org.opensearch.dataprepper.model.types.ByteCount;
+import org.opensearch.dataprepper.plugins.codec.CompressionOption;
+import org.opensearch.dataprepper.plugins.otel.codec.OTelOutputFormat;
+import org.opensearch.dataprepper.plugins.server.RetryInfoConfig;
+import org.opensearch.dataprepper.model.configuration.PluginModel;
+
+public class OTLPSourceConfig {
+  static final String REQUEST_TIMEOUT = "request_timeout";
+  static final String PORT = "port";
+  static final String LOGS_PATH = "logs_path";
+  static final String METRICS_PATH = "metrics_path";
+  static final String TRACES_PATH = "traces_path";
+  static final String SSL = "ssl";
+  static final String LOGS_OUTPUT_FORMAT = "logs_output_format";
+  static final String METRICS_OUTPUT_FORMAT = "metrics_output_format";
+  static final String TRACES_OUTPUT_FORMAT = "traces_output_format";
+  static final String USE_ACM_CERT_FOR_SSL = "useAcmCertForSSL";
+  static final String ACM_CERT_ISSUE_TIME_OUT_MILLIS = "acmCertIssueTimeOutMillis";
+  static final String HEALTH_CHECK_SERVICE = "health_check_service";
+  static final String PROTO_REFLECTION_SERVICE = "proto_reflection_service";
+  static final String SSL_KEY_CERT_FILE = "sslKeyCertChainFile";
+  static final String SSL_KEY_FILE = "sslKeyFile";
+  static final String ACM_CERT_ARN = "acmCertificateArn";
+  static final String ACM_PRIVATE_KEY_PASSWORD = "acmPrivateKeyPassword";
+  static final String AWS_REGION = "awsRegion";
+  static final String THREAD_COUNT = "thread_count";
+  static final String MAX_CONNECTION_COUNT = "max_connection_count";
+  static final String ENABLE_UNFRAMED_REQUESTS = "unframed_requests";
+  static final String COMPRESSION = "compression";
+  static final String RETRY_INFO = "retry_info";
+  static final int DEFAULT_REQUEST_TIMEOUT_MS = 10000;
+  static final int DEFAULT_PORT = 21893;
+  static final int DEFAULT_THREAD_COUNT = 200;
+  static final int DEFAULT_MAX_CONNECTION_COUNT = 500;
+  static final boolean DEFAULT_SSL = true;
+  static final boolean DEFAULT_ENABLED_UNFRAMED_REQUESTS = false;
+  static final boolean DEFAULT_HEALTH_CHECK = false;
+  static final boolean DEFAULT_PROTO_REFLECTION_SERVICE = false;
+  static final boolean DEFAULT_USE_ACM_CERT_FOR_SSL = false;
+  static final int DEFAULT_ACM_CERT_ISSUE_TIME_OUT_MILLIS = 120000;
+  private static final String S3_PREFIX = "s3://";
+  static final String UNAUTHENTICATED_HEALTH_CHECK = "unauthenticated_health_check";
+
+  @JsonProperty(REQUEST_TIMEOUT)
+  private int requestTimeoutInMillis = DEFAULT_REQUEST_TIMEOUT_MS;
```

Comment:
Have this take a Data Prepper `Duration` instead. This is our new way to handle timeouts.

Here is an example:

https://github.com/opensearch-project/data-prepper/blob/d76fe1799e795c256bf7cb932173e2bec123631f/data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/cloudwatch_logs/config/ThresholdConfig.java#L41-L44